### PR TITLE
GET /api/topics complete

### PR DESCRIPTION
**Pull request by skelbon**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    test('should return the topics from the table in nc_news_test db with both properties, slug and description', ()=>{
+        return request(app)
+        .get('/api/topics')
+        .then(({body})=>{
+            const topicsArr = body.topics
+            const exampleTopic = {
+                description: 'what books are made of',
+                slug: 'paper'
+              }
+            topicsArr.forEach((topic)=> {
```
Comment (pippaamy) [inline __tests__/app.test.js]: I would check the length here too

File: app.controller.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const {fetchTopics} = require('./app.model')
+const endpoints = require('./endpoints.json')
+
+exports.getTopics = (req, res, next)=>{
+    fetchTopics().then((topics)=>{
+        if (!topics[0]) {
```
Comment (pippaamy) [inline app.controller.js]: I think this is a bit sad pathy, but I can see why you did it

File: app.controller.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+            res.status(200).send({message: 'The topics table is empty', data: topics})
+        }
+        res.status(200).send({topics})
+    })
+}
+
+exports.getApiDescription = (req,res, next)=>{
```
Comment (pippaamy) [inline app.controller.js]: this should be part of a separate ticket

File: app.js [hunk 4]
```diff
@@ -1,9 +1,15 @@
-const {getTopics} = require('./app.controller')
+const {getTopics, getApiDescription} = require('./app.controller')
 const express = require('express')
 const app = express()
 
 app.get('/api/topics', getTopics)
 
+app.get('/api', getApiDescription )
```
Comment (pippaamy) [inline app.js]: this should be in a different branch

File: app.js [hunk 5]
```diff
@@ -1,9 +1,15 @@
...
+
+app.all('/*', (req, res, next)=>{
```
Comment (pippaamy) [inline app.js]: nice!

File: app.model.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+const db = require('./db/connection')
+
+
+exports.fetchTopics = ()=>{
+    return db.query(`SELECT * FROM topics;`).then(({rows})=>{
```
Comment (pippaamy) [inline app.model.js]: great!

**Review by pippaamy — COMMENTED** [review]
good job, make sure to keep your tickets separate

File: __tests__/app.test.js (hunk 1 above)
Comment (skelbon) [reply __tests__/app.test.js]: I  already checked the length in the test above it. Do you think I should check it within the same test?

File: app.controller.js (hunk 2 above)
Comment (skelbon) [reply app.controller.js]: Not sure what you mean by this - should I be doing this in error handling perhaps?

File: app.controller.js (hunk 3 above)
Comment (skelbon) [reply app.controller.js]: noted

File: app.js (hunk 4 above)
Comment (skelbon) [reply app.js]: noted

File: app.js (hunk 5 above)
Comment (skelbon) [reply app.js]: Thank you

File: app.model.js (hunk 6 above)
Comment (skelbon) [reply app.model.js]: Great

**Pull request merged**